amr-wind/turbulence/RANS/KLAxell.cpp

```diff
@@ -0,0 +1,349 @@
+                            std::pow(tke_arr(i, j, k) / epsilon, 2) *
+                            stratification;
+                        Rt = (Rt > Rtc)
+                                 ? Rt
+                                 : std::max(
+                                       Rt, Rt - std::pow(Rt - Rtc, 2) /
+                                                    (Rt + Rtmin - 2 * Rtc));
+                        tlscale_arr(i, j, k) =
+                            (stratification > 0)
+                                ? std::sqrt(
+                                      std::pow(lscale_s * lscale_b, 2) /
+                                      (std::pow(lscale_s, 2) +
+                                       std::pow(lscale_b, 2)))
+                                : lscale_s *
+                                      std::sqrt(
+                                          1.0 -
+                                          std::pow(Cmu, 6.0) *
+                                              std::pow(Cb_unstable, -2.0) * Rt);
+                        tlscale_arr(i, j, k) =
+                            (stratification > 0)
+                                ? std::min(
+                                      tlscale_arr(i, j, k),
+                                      std::sqrt(
+                                          Cmu * tke_arr(i, j, k) /
+                                          stratification))
+                                : tlscale_arr(i, j, k);
+                        tlscale_arr(i, j, k) = (std::abs(surf_flux) < 1e-5 &&
+                                                z <= lengthscale_switch)
+                                                   ? lscale_s
+                                                   : tlscale_arr(i, j, k);
+                        Rt = (std::abs(surf_flux) < 1e-5 &&
+                              z <= lengthscale_switch)
+                                 ? 0.0
+                                 : Rt;
+                        const amrex::Real Cmu_Rt =
+                            (Cmu + 0.108 * Rt) /
+                            (1 + 0.308 * Rt + 0.00837 * std::pow(Rt, 2));
+                        mu_arr(i, j, k) = rho_arr(i, j, k) * Cmu_Rt *
+                                          tlscale_arr(i, j, k) *
+                                          std::sqrt(tke_arr(i, j, k)) *
+                                          (1 - blank_arr(i, j, k));
+                        const amrex::Real Cmu_prime_Rt = Cmu / (1 + 0.277 * Rt);
+                        const amrex::Real muPrime =
+                            rho_arr(i, j, k) * Cmu_prime_Rt *
+                            tlscale_arr(i, j, k) * std::sqrt(tke_arr(i, j, k)) *
+                            (1 - blank_arr(i, j, k));
+                        buoy_prod_arr(i, j, k) = -muPrime * stratification;
+                        shear_prod_arr(i, j, k) *=
+                            shear_prod_arr(i, j, k) * mu_arr(i, j, k);
+                    });
+            } else {
```

Comment:
There is so much duplicated code between this else branch and the if branch, can we refactor to avoid that?

Reply:
This was done to ensure that when we need to support terrain, we do not have so many teary statements.

Reply:
The issue is most of the terms depend on length scale and length scale has a wall-distance function which is dependent on terrain or no terrain. So 90% of the variables change.